### PR TITLE
Android native geolocation

### PR DESCRIPTION
**Pull request by 007-surajit**
### Platforms affected
Android

### Motivation and Context
Fixes open issue GH-122 and GH-166 on Android platforms.

### Description
Added fused location provider to obtain accurate location.

### Testing
Tested with Cordova android 8.0.0 on Android devices running Lollipop(5.0) and Marshmallow(6.0)

### Checklist

- [x]  I've run the tests to see all new and existing tests pass
- [x] I added automated test coverage as appropriate for this change
- [x] Commit is prefixed with `(platform)` if this change only applies to one platform (e.g. `(android)`)
- [x] If this Pull Request resolves an issue, I linked to the issue in the text above (and used the correct [keyword to close issues using keywords](https://help.github.com/articles/closing-issues-using-keywords/))
- [x] I've updated the documentation if necessary

### CODE REVIEW

**Comment by 007-surajit** [comment]
<config-file` parent="/resources" target="./res/values/strings.xml">
        <string name="save_url">https://something.com/location/save</string>
  </config-file>

**Comment by janpio** [comment]
> This line needs to be added in config.xml to specify url to save location.

That looks a bit like a second feature that is included in the actual Pull Request, could that be? If so, it would probably be better to exclude that here first and create another PR that adds that second functionality so we can properly test and review the first PR.

**Comment by janpio** [comment]
(Close and reopen to restart CI)

**Comment by janpio** [comment]
Android app doesn#t build:

```
> Task :app:compileDebugJavaWithJavac FAILED
/tmp/tmp-4024nDQ7bthWZ030/platforms/android/app/src/main/java/org/apache/cordova/geolocation/FusedLocationHelper.java:48: error: package com.android.volley.toolbox does not exist
import com.android.volley.toolbox.JsonObjectRequest;
                                 ^
/tmp/tmp-4024nDQ7bthWZ030/platforms/android/app/src/main/java/org/apache/cordova/geolocation/FusedLocationHelper.java:50: error: package com.android.volley does not exist
import com.android.volley.Request;
                         ^
/tmp/tmp-4024nDQ7bthWZ030/platforms/android/app/src/main/java/org/apache/cordova/geolocation/FusedLocationHelper.java:51: error: package com.android.volley does not exist
import com.android.volley.Response;
                         ^
/tmp/tmp-4024nDQ7bthWZ030/platforms/android/app/src/main/java/org/apache/cordova/geolocation/FusedLocationHelper.java:52: error: package com.android.volley does not exist
import com.android.volley.VolleyError;
                         ^
/tmp/tmp-4024nDQ7bthWZ030/platforms/android/app/src/main/java/org/apache/cordova/geolocation/FusedLocationHelper.java:53: error: package com.android.volley.Response does not exist
import com.android.volley.Response.ErrorListener;
                                  ^
/tmp/tmp-4024nDQ7bthWZ030/platforms/android/app/src/main/java/org/apache/cordova/geolocation/MySingleton.java:4: error: package com.android.volley does not exist
import com.android.volley.Request;
                         ^
/tmp/tmp-4024nDQ7bthWZ030/platforms/android/app/src/main/java/org/apache/cordova/geolocation/MySingleton.java:5: error: package com.android.volley does not exist
import com.android.volley.RequestQueue;
                         ^
/tmp/tmp-4024nDQ7bthWZ030/platforms/android/app/src/main/java/org/apache/cordova/geolocation/MySingleton.java:6: error: package com.android.volley.toolbox does not exist
import com.android.volley.toolbox.Volley;
                                 ^
/tmp/tmp-4024nDQ7bthWZ030/platforms/android/app/src/main/java/org/apache/cordova/geolocation/MySingleton.java:10: error: cannot find symbol
    private RequestQueue requestQueue;
            ^
  symbol:   class RequestQueue
  location: class MySingleton
/tmp/tmp-4024nDQ7bthWZ030/platforms/android/app/src/main/java/org/apache/cordova/geolocation/MySingleton.java:25: error: cannot find symbol
    public RequestQueue getRequestQueue() {
           ^
  symbol:   class RequestQueue
  location: class MySingleton
/tmp/tmp-4024nDQ7bthWZ030/platforms/android/app/src/main/java/org/apache/cordova/geolocation/MySingleton.java:34: error: cannot find symbol
    public <T> void addToRequestQueue(Request<T> req) {
                                      ^
  symbol:   class Request
  location: class MySingleton
/tmp/tmp-4024nDQ7bthWZ030/platforms/android/app/src/main/java/org/apache/cordova/geolocation/FusedLocationHelper.java:187: error: cannot find symbol
            JsonObjectRequest jsonObjectRequest = new JsonObjectRequest(Request.Method.POST, locationApiUrl, obj,
            ^
  symbol:   class JsonObjectRequest
  location: class FusedLocationHelper
/tmp/tmp-4024nDQ7bthWZ030/platforms/android/app/src/main/java/org/apache/cordova/geolocation/FusedLocationHelper.java:187: error: cannot find symbol
            JsonObjectRequest jsonObjectRequest = new JsonObjectRequest(Request.Method.POST, locationApiUrl, obj,
                                                      ^
  symbol:   class JsonObjectRequest
  location: class FusedLocationHelper
/tmp/tmp-4024nDQ7bthWZ030/platforms/android/app/src/main/java/org/apache/cordova/geolocation/FusedLocationHelper.java:187: error: package Request does not exist
            JsonObjectRequest jsonObjectRequest = new JsonObjectRequest(Request.Method.POST, locationApiUrl, obj,
                                                                               ^
/tmp/tmp-4024nDQ7bthWZ030/platforms/android/app/src/main/java/org/apache/cordova/geolocation/FusedLocationHelper.java:188: error: package Response does not exist
                    new Response.Listener<JSONObject>() {
                                ^
/tmp/tmp-4024nDQ7bthWZ030/platforms/android/app/src/main/java/org/apache/cordova/geolocation/FusedLocationHelper.java:195: error: package Response does not exist
                    new Response.ErrorListener() {
                                ^
/tmp/tmp-4024nDQ7bthWZ030/platforms/android/app/src/main/java/org/apache/cordova/geolocation/MySingleton.java:29: error: cannot find symbol
            requestQueue = Volley.newRequestQueue(ctx.getApplicationContext());
                           ^
  symbol:   variable Volley
  location: class MySingleton
```

File: src/android/FusedLocationHelper.java [hunk 1]
```diff
@@ -166,8 +169,38 @@ public void onConnectionFailed(@NonNull ConnectionResult connectionResult) {
...
+                obj.put("user_id", 1);
+                obj.put("latitude", location.getLatitude());
+                obj.put("longitude", location.getLongitude());
+                obj.put("timestamp", location.getTime());
+
+            } catch (JSONException e) {
+                throw new RuntimeException(e);
+            }
+
+            JsonObjectRequest jsonObjectRequest = new JsonObjectRequest(Request.Method.POST, locationApiUrl, obj,
```
Comment (breautek) [inline src/android/FusedLocationHelper.java]: Can you describe why this appears to be sending latitude/longitude information to some server via an HTTP Post request?

What is this for?

Comment (janpio) [reply src/android/FusedLocationHelper.java]: I think this is the second feature I referred to here: https://github.com/apache/cordova-plugin-geolocation/pull/168#issuecomment-509143411

Comment (breautek) [reply src/android/FusedLocationHelper.java]: What's the purpose of this feature? If the developer wants to send GPS to a server, wouldn't it make more sense to just simply use  javascript XHR?

The plugin responsibility should just be an API to provide GPS results to the app... anything more is probably out of scope of what the plugin should be responsible for.

Comment (janpio) [reply src/android/FusedLocationHelper.java]: I don't know, as it isn't described here. Even if there is a use for it, this should be moved to a separate PR (see my comment).

**Comment by breautek** [comment]
Closing due to staleness.

**Pull request closed without merging**